src/Yaapii.Atoms/Map/LazyDict.cs

```diff
@@ -60,36 +69,60 @@ public LazyDict(IEnumerable<IKvp> kvps)
                     }
                     return dict;
                 });
+            this.anyValueIsLazy = new Sticky<bool>(() =>
+            {
+                return new Ternary<IFail, bool>(
```

Comment:
It seems easier to me if we combine the mapped with a Filtered and ask its length.

But for performance reasons, I suggest we simply move an enumerator and break if we find a lazy kvp. That way we reduce the test to one iteration which we also might not need to fully finish.